iceoryx_hoofs/test/moduletests/test_relative_pointer.cpp

```diff
@@ -243,158 +279,162 @@ TYPED_TEST(RelativePointer_test, AssignmentOperatorTests)
+
+    int offset = SHARED_MEMORY_SIZE / 2U;
+    // NOLINTJUSTIFICATION Pointer arithmetic needed for tests
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic, cppcoreguidelines-pro-type-reinterpret-cast)
+    auto* addressAtOffset = reinterpret_cast<TypeParam*>(ptr + offset);
+    RelativePointer<TypeParam> rp2(addressAtOffset, iox::rp::BaseRelativePointer::id_t{1U});
+    EXPECT_EQ(BaseRelativePointer::getOffset(iox::rp::BaseRelativePointer::id_t{1U}, addressAtOffset), offset);
 }
 
-TYPED_TEST(RelativePointer_test, getPtr)
+TYPED_TEST(RelativePointer_test, GetPtrReturnsAddressWithCorrectOffset)
 {
     ::testing::Test::RecordProperty("TEST_ID", "4fadf89f-69c0-4058-8995-a98e2e3334b2");
-    auto ptr = this->partitionPtr(0);
-    auto typedPtr = reinterpret_cast<TypeParam*>(ptr);
-    RelativePointer<TypeParam> rp1(typedPtr, 1);
-    EXPECT_EQ(rp1.registerPtr(1, typedPtr), true);
-    EXPECT_EQ(BaseRelativePointer::getPtr(1, 0), ptr);
+    auto* ptr = this->partitionPtr(0U);
+    auto* typedPtr = static_cast<TypeParam*>(static_cast<void*>(this->partitionPtr(0U)));
+    RelativePointer<TypeParam> rp1(typedPtr, iox::rp::BaseRelativePointer::id_t{1U});
+    EXPECT_EQ(rp1.registerPtr(iox::rp::BaseRelativePointer::id_t{1U}, typedPtr), true);
+    EXPECT_EQ(BaseRelativePointer::getPtr(iox::rp::BaseRelativePointer::id_t{1U}, 0), typedPtr);
 
     uint64_t offset = SHARED_MEMORY_SIZE / 2U;
-    auto addressAtOffset = reinterpret_cast<TypeParam*>(ptr + offset);
-    RelativePointer<TypeParam> rp2(addressAtOffset, 1);
-    EXPECT_EQ(BaseRelativePointer::getPtr(1, offset), addressAtOffset);
+    // NOLINTJUSTIFICATION Pointer arithmetic needed for tests
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic, cppcoreguidelines-pro-type-reinterpret-cast)
+    auto* addressAtOffset = reinterpret_cast<TypeParam*>(ptr + offset);
+    RelativePointer<TypeParam> rp2(addressAtOffset, iox::rp::BaseRelativePointer::id_t{1});
+    EXPECT_EQ(BaseRelativePointer::getPtr(iox::rp::BaseRelativePointer::id_t{1U}, offset), addressAtOffset);
 }
 
-TYPED_TEST(RelativePointer_test, registerPtr)
+TYPED_TEST(RelativePointer_test, RegisteringAndUnregisteringRelativePointerWorks)
 {
     ::testing::Test::RecordProperty("TEST_ID", "3f08ab46-c778-468a-bab1-ecd71aa800f4");
-    auto ptr = this->partitionPtr(0);
-    auto typedPtr = reinterpret_cast<TypeParam*>(ptr);
+    auto* typedPtr = static_cast<TypeParam*>(static_cast<void*>(this->partitionPtr(0U)));
 
-    RelativePointer<TypeParam> rp1(typedPtr, 1);
+    RelativePointer<TypeParam> rp1(typedPtr, iox::rp::BaseRelativePointer::id_t{1U});
 
-    EXPECT_EQ(rp1.registerPtr(1, typedPtr), true);
-    EXPECT_EQ(rp1.registerPtr(1, typedPtr), false);
-    EXPECT_EQ(rp1.unregisterPtr(1), true);
-    EXPECT_EQ(rp1.registerPtr(1, typedPtr), true);
+    EXPECT_EQ(rp1.registerPtr(iox::rp::BaseRelativePointer::id_t{1U}, typedPtr), true);
```

Comment:
If this is deemed to verbose we could use an alias in the test file (`id_t` or `Id` etc.). I would prefer this.

Reply:
Once, the AUTOSAR rule violations get fixed `id_t` needs to be renamed anyway. I would suggest `identifier_t`.